emigration.lua

```diff
@@ -56,25 +56,27 @@ function desert(u,method,civ)
 end
 
 function canLeave(unit)
+    if not unit.status.current_soul then
+        return false
+    end
+    
     for _, skill in pairs(unit.status.current_soul.skills) do
         if skill.rating > 14 then return false end
     end
-    if unit.flags1.caged
-        or unit.race ~= df.global.ui.race_id
-        or unit.civ_id ~= df.global.ui.civ_id
-        or dfhack.units.isDead(unit)
-        or dfhack.units.isOpposedToLife(unit)
-        or unit.flags1.merchant
-        or unit.flags1.diplomat
-        or unit.flags1.chained
-        or dfhack.units.getNoblePositions(unit) ~= nil
-        or unit.military.squad_id ~= -1
-        or dfhack.units.isCitizen(unit)
-        or dfhack.units.isSane(unit)
-        or unit.profession ~= 103
-        or not dfhack.units.isDead(unit)
-            then return false end
-    return true
+    
+    return dfhack.units.isOwnRace(unit) and  --  Doubtful check. naturalized citizens
+           dfhack.units.isOwnCiv(unit) and   --  might also want to leave.
+           not dfhack.units.isActive(unit) and
```

Comment:
Isn't this still reversed? Active units can leave, but inactive units can't.

Reply:
Yes it is...